bluemira/base/parameter_frame/_frame.py

```diff
@@ -38,11 +49,15 @@ class AnotherFrame(NewParameterFrame):
 
     """
 
-    def __eq__(self, __o: object) -> bool:
-        """Check whether two ParameterFrames are equal."""
-        if not isinstance(__o, NewParameterFrame):
-            return NotImplemented
-        return self.__dataclass_fields__ == __o.__dataclass_fields__
+    def __iter__(self) -> Generator[NewParameter, None, None]:
+        """
+        Iterate over this frame's parameters.
+
+        The order is based on the order in which the parameters were
+        declared.
+        """
+        for field in fields(self):
+            yield getattr(self, field.name)
```

Comment:
The diff here is confusing I think; `__iter__` is not replacing `__eq__`. We don't need to define `__eq__`, because we get it for free from `dataclass` - see 80c1cbd18f68a41741c0da5243c8b9c4a213add2. 

The `__iter__` method is just an ease of use thing. I'm using it in the [PROCESS solver's teardown](https://github.com/Fusion-Power-Plant-Framework/bluemira/pull/1489/commits/35ca25f2869fdc5f25cc4438fcd23f2cfc653b67#diff-2ebff4dc0151e68b5144197de56f0b29955270bd3365e93d56e42005cff30a0eR157) to loop through each parameter, and get a dict that maps the parameter name to its unit.

Reply:
Fair enough!